test/tests/partitioners/single_rank_partitioner/tests

```diff
@@ -0,0 +1,20 @@
+[Tests]
+  issues = '#18729'
+  design = 'partitioners/SingleRankPartitioner.md'
+  requirement = 'The system shall error if an out of range rank is given to SingleRankPartitioner'
```

Comment:
Ah - this won't quite work though. You need to use the detail syntax; you cannot have duplicate requirements for a single test

Reply:
Yeah - just me being hasty.  Let me fix it.